### ISSUE TITLE
SAT cannot read some models/iterations due to error

### ISSUE THREAD

**Issue by cozminvelciu** [RHEAGROUP/COMET-Server-Administration-Tool#104]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-SDK
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-Server-Administration-Tool/issues) and [closed](https://github.com/RHEAGROUP/CDP4-Server-Administration-Tool/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
During the CDF production migration workshop, SAT could not read some of the models (confidential names list can be found on the [ESA gitlab issue](https://gitlab.esa.int/ocdt/ocdt-portal/-/issues/4378)).

The cause is the following error:
```
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at CDP4Dal.Session.<Read>d__48.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at CDP4Dal.Session.<Read>d__48.MoveNext()<---
```

Note that this only happens when reading models from SAT, as the models can be read with other clients with no problem.
### System Configuration

- [ ] CDP4-Server-Administration-Tool version:
- [ ] Environment (Operating system, version and so on):
- [ ] .NET Framework version:
- [ ] Additional information:

**Comment by cozminvelciu** [comment]
The error occurs when trying to read the iterations of any model with more than one iteration, and it seems to be caused by:
* `System.InvalidOperationException : Collection was modified; enumeration operation may not execute.`

The problem is that the `IterationSetup` list is modified when reading an `Iteration`, which causes the `foreach` iterator to fail when trying to move to the second `Iteration`.

This can be easily fixed in the SAT by making a clone of the list of `IterationSetup`s for each `EngineeringModelSetup`, but I wonder why is the list `engineeringModelSetup.IterationSetup` modified when reading an `Iteration`.

**Comment by cozminvelciu** [comment]
The problem was caused by the fact that SAT v1.2.0 uses SDK v7.3.0 - which does not include the fix for [SAT issue #97](https://github.com/RHEAGROUP/COMET-Server-Administration-Tool/issues/97) ([SDK issue #167](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues/167)).

**Comment by cozminvelciu** [comment]
The problem was caused by the fact that SAT v1.2.0 uses SDK v7.3.0 - which does not include the fix for [SAT issue #97](https://github.com/RHEAGROUP/COMET-Server-Administration-Tool/issues/97) ([SDK issue #167](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues/167)).

We need to make a new CDP4 SDK release version, and then a new SAT release which references the new SDK version (and includes all the SAT fixes).

**Comment by cozminvelciu** [comment]
Issues added for new versions:
* SDK: [issue #187](https://github.com/RHEAGROUP/COMET-SDK-Community-Edition/issues/187)
* SAT: [issue #107](https://github.com/RHEAGROUP/COMET-Server-Administration-Tool/issues/107)
